flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -237,57 +230,73 @@ private void recoverWorkerNodesFromPreviousAttempts() throws ResourceManagerExce
 			++currentMaxAttemptId);
 	}
 
-	private void requestKubernetesPod() {
-		numPendingPodRequests++;
+	private void requestKubernetesPod(WorkerResourceSpec workerResourceSpec) {
+		final KubernetesTaskManagerParameters parameters =
+			createKubernetesTaskManagerParameters(workerResourceSpec);
+
+		final KubernetesPod taskManagerPod =
+			KubernetesTaskManagerFactory.createTaskManagerComponent(parameters);
+		kubeClient.createTaskManagerPod(taskManagerPod);
```

Comment:
This is also unrelated to this PR. The naming is a bit unfortunate. We have a `createTaskManagerComponent` which creates a `KubernetesPod` and a `createTaskManagerPod` which does not seem to create/return something. I think `createXYZ` should create `XYZ` and return it. Hence, the first method should be called `createKubernetesPod` and the second could be called `startKubernetesPod` or `startTaskManagerPod`.

Reply:
Sorry to jump to here. I want to add some additional explanations here. 

Why i name it to `createTaskManagerPod` instead of `startTaskManagerPod`?
* we usually use `create` in K8s commands(e.g. `kubectl create -f taskmanager.yaml`). The start process will be done internally and automatically in K8s `kubelet`. 

Why do not return a `Pod` for `createTaskManagerPod`?
* Of cause we could return a created `Pod`. However, it is not a final or stable state. Since the K8s will do some updates for the `Pod`, the ip address, the status. So i do not return the `Pod` to avoid it may be used incorrectly. 

For the `KubernetesJobManagerFactory#createJobManagerComponent` and `KubernetesTaskManagerFactory#createTaskManagerComponent`, maybe we could rename them to `KubernetesJobManagerFactory#createJobManagerSpecification` and `KubernetesTaskManagerFactory#createTaskManagerPod`.

Reply:
I think my main problem is that we have two `createXYZ` methods where the one method is creating an object and the other method doing some side effects. Since they are doing different things, they should not have the same prefix. Which one starts with `createXYZ` does not matter but I would like to break up the seemingly symmetry of these methods which does not exist.

Moreover, I think it is a good practice to include the name of the type a method instantiates in the method name: `KubernetesPod create/instantiate/buildKubernetesPod()` or `JobManagerPodSpecification create/instantiate/buildJobManagerPodSpecification`.

Reply:
Thanks for your explanation. From your suggestion guideline, the following two interfaces are already in good shape.
```
// Create the Master components, this can include the Deployment, the ConfigMap(s), and the Service(s).
void createJobManagerComponent(KubernetesJobManagerSpecification kubernetesJMSpec);
CompletableFuture<Void> createTaskManagerPod(KubernetesPod kubernetesPod);
```

And the following two methods could be renamed.
```
public static KubernetesJobManagerSpecification createJobManagerComponent(KubernetesJobManagerParameters kubernetesJobManagerParameters)
public static KubernetesPod createTaskManagerComponent(KubernetesTaskManagerParameters kubernetesTaskManagerParameters)
```
===>
```
public static KubernetesJobManagerSpecification buildKubernetesJobManagerSpecification(KubernetesJobManagerParameters kubernetesJobManagerParameters)
public static KubernetesPod buildTaskManagerKubernetesPod(KubernetesTaskManagerParameters kubernetesTaskManagerParameters)
```

Reply:
Jup, this would be a good improvement.